src/elements/fields/SignatureField/components/SignatureModal.tsx

```diff
@@ -0,0 +1,375 @@
+import React, { useCallback, useRef, useState } from 'react';
+import SignatureCanvas, { SignatureCanvasProps } from './SignatureCanvas';
+import { CloseIcon } from '../../../components/icons';
+import { dataURLToFile } from '../../../../utils/image';
+import html2canvas from 'html2canvas';
+import debounce from 'lodash.debounce';
+
+type SignatureModalProps = SignatureCanvasProps & {
+  show: boolean;
+  setShow: (val: boolean) => void;
+  returnFile?: boolean;
+};
+
+function SignatureModal(props: SignatureModalProps) {
+  const {
+    show = false,
+    setShow = () => {},
+    fieldKey,
+    responsiveStyles,
+    onClear = () => {},
+    onEnd = () => {}
+  } = props;
+
+  const [drawSignature, setDrawSignature] = useState(false);
+  const [fullName, setFullName] = useState('');
+  const [isLoading, setLoading] = useState(false);
+  const [signatureFile, setSignatureFile] = useState<File>();
+  const [signatureImgData, setSignatureImgData] = useState('');
+  const previewRef = useRef<HTMLDivElement>(null);
+
+  const resetState = () => {
+    setSignatureFile(undefined);
+    setSignatureImgData('');
+    setFullName('');
+    setDrawSignature(false);
+  };
+
+  const generateSignature = () => {
+    if (previewRef.current) {
+      html2canvas(previewRef.current, { backgroundColor: null }).then(
+        (canvas) => {
+          const imgData = canvas.toDataURL('image/png');
+          const imgFile = dataURLToFile(imgData, `${fieldKey}.png`);
+
+          setSignatureImgData(imgData);
+          setSignatureFile(imgFile);
+          setLoading(false);
+        }
+      );
+    } else {
+      setLoading(false);
+    }
+  };
+
+  const debounceGenerateSignature = useCallback(
+    debounce(generateSignature, 1000),
+    []
+  );
+
+  const handleCancel = () => {
+    setShow(false);
+    resetState();
+  };
+
+  const handleSubmit = () => {
+    onEnd(signatureFile);
```

Comment:
Edge case: With the old signature control, it was possible to unsign in that the clear icon would appear on hover and you could remove the signature.  Clearing would clear the value of the signature field.  It seems like that is not present here and there is no way to do it.  I think we need it.  Should be as simple as clicking sign with no name entered or no signature, clears the value.

Reply:
And maybe the Sign label become "Unsign" until something is entered?